tally/include/tally/buckets.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include <chrono>
+#include <vector>
+
+#include "tally/buckets_iterator.h"
+#include "tally/src/buckets_calculator.h"
+
+namespace tally {
+
+class Buckets {
+ public:
+  // LinearValues constructs a linear sequence of `num` Value buckets beginning
+  // at `start` and incrementing by `width` each time.
+  static Buckets LinearValues(double start, double width, uint64_t num);
```

Comment:
nit: change method naming convention to `lowerCase`

Reply:
hm actually, how does the style guide reconcile situations where begin()/end() etc have to follow the stl pattern and ctors have to be the same as the class name?

Reply:
I have no idea, I was trying to following the Google style guide which says use lowercase for getters and setters and uppercase otherwise except where I had to implement an interface or the standard library has other conventions.

Reply:
As noted, not a fan of Google style guide.